### PR TITLE
1.0.0

### LINKED ISSUE #11: Bug: Cannot read property 'date' of undefined
OS: Windows
Node Ver: 10.x

Console outputs: 

````cmd
C:\Users\Luke Shiels\gh-contributions>npm start

> gh-contributions@0.0.0 start C:\Users\Luke Shiels\gh-contributions
> node server.js

Listening on 9000
/css/gh-style.css - Not Modified
/js/jquery.min.js - Not Modified
/bootstrap/js/bootstrap.min.js - Not Modified
/js/index.js - Not Modified
/bootstrap/css/bootstrap.min.css - Not Modified
/css/github.css - Not Modified

C:\Users\Luke Shiels\gh-contributions\contributions.js:60
            options.dates.push(year[p.x - 1][p.y - 1].date);
                                                     ^
TypeError: Cannot read property 'date' of undefined
    at Object.module.exports.getRepo (C:\Users\Luke Shiels\gh-contributions\co
ributions.js:60:54)
    at C:\Users\Luke Shiels\gh-contributions\server.js:53:27
    at IncomingMessage.<anonymous> (C:\Users\Luke Shiels\gh-contributions\serv
.js:107:13)
    at IncomingMessage.EventEmitter.emit (events.js:92:17)
    at _stream_readable.js:920:16
    at process._tickCallback (node.js:415:13)

npm ERR! gh-contributions@0.0.0 start: `node server.js`
npm ERR! Exit status 8
npm ERR!
npm ERR! Failed at the gh-contributions@0.0.0 start script.
npm ERR! This is most likely a problem with the gh-contributions package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node server.js
npm ERR! You can get their info via:
npm ERR!     npm owner ls gh-contributions
npm ERR! There is likely additional logging output above.
npm ERR! System Windows_NT 6.1.7600
npm ERR! command "C:\\Program Files\\nodejs\\\\node.exe" "C:\\Program Files\\n
ejs\\node_modules\\npm\\bin\\npm-cli.js" "start"
npm ERR! cwd C:\Users\Luke Shiels\gh-contributions
npm ERR! node -v v0.10.26
npm ERR! npm -v 1.4.3
npm ERR! code ELIFECYCLE
npm ERR!
npm ERR! Additional logging details can be found in:
npm ERR!     C:\Users\Luke Shiels\gh-contributions\npm-debug.log
npm ERR! not ok code 0

C:\Users\Luke Shiels\gh-contributions>

````

The client on the browser just says: `Error: undefined`.

### LINKED ISSUE #16: Repo not generated after clicking 'Generate repository from data'
OS X Mavericks 10.9.2
Node version v0.10.28
Chrome v34.0.1847.131

No repo, only one README.md with 'hello wolrd' and a long list of integers.

Console:

```sh
SITYAGI-M-V03U:repos siddharth_tyagi$ ls -latr
total 88
-rw-r--r--  1 siddharth_tyagi  staff     14 Apr 14 00:24 .gitignore
drwxr-xr-x  7 siddharth_tyagi  staff    238 Apr 14 00:24 ..
drwxr-xr-x  3 siddharth_tyagi  staff    102 May  6 00:07 rtgmnbzhtxfxbt9
-rw-r--r--  1 siddharth_tyagi  staff  18872 May  6 00:21 rtgmnbzhtxfxbt9.zip
-rw-r--r--  1 siddharth_tyagi  staff  18817 May  6 00:24 kdgz54mh2jfw29.zip
drwxr-xr-x  3 siddharth_tyagi  staff    102 May  6 00:27 kdgz54mh2jfw29
drwxr-xr-x  3 siddharth_tyagi  staff    102 May  6 00:27 public
drwxr-xr-x  8 siddharth_tyagi  staff    272 May  6 00:27 .
SITYAGI-M-V03U:repos siddharth_tyagi$ ls public/repos/
rtgmnbzhtxfxbt9
SITYAGI-M-V03U:repos siddharth_tyagi$ ls public/repos/rtgmnbzhtxfxbt9/
README.md
SITYAGI-M-V03U:repos siddharth_tyagi$ unzip kdgz54mh2jfw29.zip 
Archive:  kdgz54mh2jfw29.zip
   creating: public/repos/kdgz54mh2jfw29/
  inflating: public/repos/kdgz54mh2jfw29/README.md  
SITYAGI-M-V03U:repos siddharth_tyagi$ ls
kdgz54mh2jfw29      kdgz54mh2jfw29.zip  public          rtgmnbzhtxfxbt9     rtgmnbzhtxfxbt9.zip
SITYAGI-M-V03U:repos siddharth_tyagi$ ls -latr public/repos/kdgz54mh2jfw29/
total 80
drwxr-xr-x  3 siddharth_tyagi  staff    102 May  6 00:09 .
-rw-r--r--  1 siddharth_tyagi  staff  40004 May  6 00:24 README.md
drwxr-xr-x  4 siddharth_tyagi  staff    136 May  6 00:27 ..
SITYAGI-M-V03U:repos siddharth_tyagi$ 
```

README.md contents:

Hello Wolrd
13290
17833
22376
26919
15532
20075
8689
13232
17775
22318
26861
31404
20017
7722
12265

### LINKED ISSUE #24: Provide an installer for Windows (where curl and wget are not available)

### LINKED ISSUE #31: Problems when installing on Windows
Hi,

I am having problems installing it in Windows. I read through the installation guide and tried to download and install it through PowerShell and Cygwin. In both I get errors. 

In: * PowerShell it does not recognise the `wget` or `curl` . Also the `-qO-` in the wget command throws an error.
     * In Cygwin, the installer works, but I think there is something wrong with Node?

Thanks,

Sebie

(Also, can you tag it with "Question" ? I think that is the appropriate label..)

### PR DESCRIPTION
**Pull request by IonicaBizau**
- [x] Removed `sh` calls and scripts. Fixes #16.
 - [x] Use OS agnostic calls. Fixes #31.
 - [x] Fixes  #11. **Must be tested**.
 - [x] NPM installer. Fixes #24.
 - [x] Windows compatibilty

### CODE REVIEW

**Comment by ComFreek** [comment]
Nothing happens after clicking "Generate repository from data" except for the progress bar to be animated.

**Comment by IonicaBizau** [comment]
Any errors? Do [here](https://github.com/IonicaBizau/github-contributions/blob/b61770f252d2bca58bd8c5bd713967f2398cca36/lib/client_handler/index.js#L37) a `console.log(err || repoLink)`.

**Comment by ComFreek** [comment]
Silly me :smile: I didn't switch to the 1.0.0 branch.

Being on the 1.0.0 branch, it works like a charm except for the display on my profile. I'll try again and post a screenshot.

**Comment by IonicaBizau** [comment]
Yes, you should be on the `1.0.0` branch.

*it works like a charm except for the display on my profile* -- what do you mean?

**Comment by ComFreek** [comment]
Repo: https://github.com/ComFreek/gc
Profile: https://github.com/ComFreek?tab=contributions&from=2014-07-09

Screenshot of drawing

Data:
```
{
    "coordinates": [
        {
            "x": 10,
            "y": 1
        },
        {
            "x": 11,
            "y": 1
        },
        {
            "x": 12,
            "y": 1
        },
        {
            "x": 12,
            "y": 2
        },
        {
            "x": 12,
            "y": 3
        },
        {
            "x": 12,
            "y": 4
        },
        {
            "x": 12,
            "y": 5
        },
        {
            "x": 12,
            "y": 6
        },
        {
            "x": 12,
            "y": 7
        },
        {
            "x": 13,
            "y": 1
        },
        {
            "x": 14,
            "y": 1
        },
        {
            "x": 16,
            "y": 1
        },
        {
            "x": 16,
            "y": 2
        },
        {
            "x": 16,
            "y": 3
        },
        {
            "x": 16,
            "y": 4
        },
        {
            "x": 16,
            "y": 5
        },
        {
            "x": 16,
            "y": 6
        },
        {
            "x": 16,
            "y": 7
        },
        {
            "x": 17,
            "y": 1
        },
        {
            "x": 17,
            "y": 4
        },
        {
            "x": 17,
            "y": 7
        },
        {
            "x": 18,
            "y": 1
        },
        {
            "x": 18,
            "y": 4
        },
        {
            "x": 18,
            "y": 7
        },
        {
            "x": 19,
            "y": 1
        },
        {
            "x": 19,
            "y": 4
        },
        {
            "x": 19,
            "y": 7
        },
        {
            "x": 20,
            "y": 1
        },
        {
            "x": 20,
            "y": 4
        },
        {
            "x": 20,
            "y": 7
        },
        {
            "x": 22,
            "y": 1
        },
        {
            "x": 22,
            "y": 2
        },
        {
            "x": 22,
            "y": 3
        },
        {
            "x": 22,
            "y": 4
        },
        {
            "x": 22,
            "y": 7
        },
        {
            "x": 23,
            "y": 1
        },
        {
            "x": 23,
            "y": 4
        },
        {
            "x": 23,
            "y": 7
        },
        {
            "x": 24,
            "y": 1
        },
        {
            "x": 24,
            "y": 4
        },
        {
            "x": 24,
            "y": 7
        },
        {
            "x": 25,
            "y": 1
        },
        {
            "x": 25,
            "y": 4
        },
        {
            "x": 25,
            "y": 7
        },
        {
            "x": 26,
            "y": 1
        },
        {
            "x": 26,
            "y": 4
        },
        {
            "x": 26,
            "y": 5
        },
        {
            "x": 26,
            "y": 6
        },
        {
            "x": 26,
            "y": 7
        },
        {
            "x": 28,
            "y": 1
        },
        {
            "x": 29,
            "y": 1
        },
        {
            "x": 30,
            "y": 1
        },
        {
            "x": 30,
            "y": 2
        },
        {
            "x": 30,
            "y": 3
        },
        {
            "x": 30,
            "y": 4
        },
        {
            "x": 30,
            "y": 5
        },
        {
            "x": 30,
            "y": 6
        },
        {
            "x": 30,
            "y": 7
        },
        {
            "x": 31,
            "y": 1
        },
        {
            "x": 32,
            "y": 1
        }
    ],
    "commitsPerDay": 2
}
```

**Comment by IonicaBizau** [comment]
What if you push the repository located at `/public/repos/<random>/`? The issue is that the `.git` folder is not archived I guess.

See [this line](https://github.com/IonicaBizau/github-contributions/blob/b61770f252d2bca58bd8c5bd713967f2398cca36/lib/contributions/index.js#L132) and https://github.com/ctalkington/node-archiver/issues/110.

**Comment by ComFreek** [comment]
See for yourself :smile: 
It works

**Comment by IonicaBizau** [comment]
Awesome!

So any idea to add the hidden directories to the archive? I guess that's the issue...

**Comment by ComFreek** [comment]
I think this [line in node-archiver](https://github.com/ctalkington/node-archiver/blob/ec94f952d0f7106077789cb1073451c24b192f10/lib/util/file.js#L62) is responsible for retrieving all files.

Glob doesn't list .git (at least not by default). How to reproduce:

1. `npm install git` (anywhere)
2. `git init` (to create a git folder)
3. `node` and enter `require('glob').sync("*", {});`

**Comment by IonicaBizau** [comment]
That's a good point.

```js
> require('glob').sync(".*", {});
[ '.git' ]                       
```

**Comment by IonicaBizau** [comment]
But it would be even better to have a simple interface to push the repository on GitHub, from UI. :smile: Then, we don't need any library to zip the things.

**Comment by ComFreek** [comment]
I suggest [node-webkit](https://github.com/rogerwang/node-webkit). We could just take over all the HTML, JS and CSS for the designer.

One 'problem' though: How do we handle GitHub's authorization? It's technically possible, but do the users trust foreign code? This issue can be partly circumvented by the use of auth codes which limit the actions github-contributions can do.

**Comment by IonicaBizau** [comment]
> How do we handle GitHub's authorization?

That's a good point. If the user uses SSH, then the issue is fixed. For HTTP auth is probably a little bit tricky... We will not make any GitHub authorization, but just the authorization needed for `git push`.

I cannot see how node-webkit can help us here.

**Comment by ComFreek** [comment]
> But it would be even better to have a simple interface to push the repository on GitHub, **from UI.**
Emphasis mine. node-webkit provides a means to create a GUI with HTML/CSS/(Node)JS

SSH keys would be an inconvenience for (some) Windows users once again.

**Comment by IonicaBizau** [comment]
Oh :hankey:. Then the behavior will be the same: download the zip file. :smile:

File: lib/contributions/index.js [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+                        done(zipErr, repoId);
+                    });
+
+                    archive.on("error", function(err){
+                        zipErr = err;
+                    });
+
+                    archive.pipe(output);
+                    archive.bulk([
+                        { expand: true, src: ["*", ".git/**/*"], cwd: repoPath, dest: repoId }
```
Comment (IonicaBizau) [inline lib/contributions/index.js]: @ComFreek This archives the `.git` files too. Thanks @ctalkington!

The full conversation is here: https://github.com/ctalkington/node-archiver/issues/110

**Comment by IonicaBizau** [comment]
Check out [the `1.0.0` branch](https://github.com/IonicaBizau/github-contributions/tree/1.0.0). If anything should be fixed, leave a comment here.

/cc @ComFreek @silviubogan @tyagis @sebie

**Comment by sebie** [comment]
Will check this out tomorrow and tell you how it ended. Cheers for the /cc.

**Comment by IonicaBizau** [comment]
The new versions comes with a redesign, based on the CaiuSS framework: https://github.com/IonicaBizau/CaiuSS

Also, for `git` commands I use `node-gry`: https://github.com/IonicaBizau/node-gry

**Comment by IonicaBizau** [comment]
Landed on master!

https://twitter.com/IonicaBizau/status/560352971322834944

**Pull request merged**